dht.go

```diff
@@ -284,20 +284,57 @@ func (d *DHT) findNode(id string) {
 	}
 }
 
-// Run starts a DHT node. It bootstraps a routing table, if necessary, and
-// listens for incoming DHT requests.
+// Start launches the dht node. It starts a listener
+// on the desired address, then runs the main loop in a
+// separate go routine - Start replaces Run and will
+// always return, with nil if the dht successfully
+// started or with an error either. d.Stop() is expected
+// by the caller to stop the dht
+func (d *DHT) Start() (err error) {
+	if err = d.initSocket(); err == nil {
+		go d.loop()
+	}
+	return err
+}
+
+// Run launches the dht node. It starts a listener
+// on the desired address, then runs the main loop in the
+// same go routine.
+// If initSocket fails, Run returns with the error.
+// If initSocket succeeds, Run blocks until d.Stop() is called.
+// DEPRECIATED - Start should be used instead of Run
```

Comment:
I think you mean DEPRECATED: http://stackoverflow.com/questions/9208091/the-difference-between-deprecated-depreciated-and-obsolete